docs/serializer.md

```diff
@@ -0,0 +1,185 @@
+
+#### Available \*\_cached variants of scene and model loading functions to ease serializer usage and simplify code
+* `get_vertices_and_indices_cached(gvk::serializer& aSerializer, ...)`
+* `create_vertex_and_index_buffers_cached(gvk::serializer& aSerializer, ...)`
+* `get_normals_cached(gvk::serializer& aSerializer, ...)`
+* `create_normals_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_tangents_cached(gvk::serializer& aSerializer, ...)`
+* `create_tangents_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_bitangents_cached(gvk::serializer& aSerializer, ...)`
+* `create_bitangents_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_colors_cached(gvk::serializer& aSerializer, ...)`
+* `create_colors_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_bone_weights_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_weights_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_weights_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_bone_indices_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_indices_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_bone_indices_for_single_target_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_indices_for_single_target_buffer_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_bone_indices_for_single_target_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `create_bone_indices_for_single_target_buffer_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_2d_texture_coordinates_cached(gvk::serializer& aSerializer, ...)`
+* `create_2d_texture_coordinates_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_2d_texture_coordinates_flipped_cached(gvk::serializer& aSerializer, ...)`
+* `create_2d_texture_coordinates_flipped_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `get_3d_texture_coordinates_cached(gvk::serializer& aSerializer, ...)`
+* `create_3d_texture_coordinates_buffer_cached(gvk::serializer& aSerializer, ...)`
+* `convert_for_gpu_usage_cached(gvk::serializer& aSerializer, ...)`
+
+
+## Custom type serialization
+Every type needs a special template function which defines how a type is serialized. For often used types such as `glm::vec3`, `glm::mat4`, etc. and various `std` types this special functions are already defined in [`serializer.hpp`](https://github.com/cg-tuwien/Gears-Vk/blob/master/framework/include/serializer.hpp). To add serialization ability to a custom type, a **serialize** function template in the same namespace as the custom type must be created. The function template takes two arguments, an **Archive** and your custom **Type** as reference, and passes every member of the **Type** to the **Archive**. If one of the members is also a custom type, a separate `serialize` function template must be created.
```

Comment:
```suggestion
To serialize a custom type, the custom type is required to have a specialized overload to the `gvk::serialize` template function which defines how a type is serialized. For commonly used types such as `glm::vec3`, `glm::mat4`, etc. and various types in the `std` namespace such `gvk::serialize` are already defined in [`serializer.hpp`](https://github.com/cg-tuwien/Gears-Vk/blob/master/framework/include/serializer.hpp). 

To add serialization ability to a custom type, a `gvk::serialize` function template overload in the same namespace as the custom type must be created. The function template takes two arguments, an `Archive` and the custom type's type `Type`. Both parameters must be passed by reference. If one of the members is yet another custom type, a separate `gvk::serialize` function template overload must be created.
```

Please double check if what I changed makes sense! (Not 100% sure about that. --> Especially my changes to `gvk::serialize` might be wrong. Because you indicated that the namespaces must match the custom type exactly. Sorry if I made some things more confusing! Feel free to re-write according to your judgement.) Please try to get the information given on point---as precisely as possible!